Tests/Functional/JobRepositoryTest.php

```diff
@@ -96,7 +96,33 @@ public function testFindStartableJob()
 
         $excludedIds = array();
         $this->assertSame($c, $this->repo->findStartableJob($excludedIds));
-        $this->assertEquals(array($b->getId()), $excludedIds);
+        $this->assertEmpty($excludedIds);
+    }
+
+    public function testFindStartableJobWithFinishedDependency()
+    {
+        $this->assertNull($this->repo->findStartableJob());
+
+        $a = new Job('a');
+        $a->setState('running');
+        $b = new Job('b');
+        $c = new Job('c');
+        $c->setState('running');
+        $c->setState('finished');
+        $d = new Job('d');
+        $b->addDependency($c);
+        $b->addDependency($d);
+
+        $this->em->persist($a);
+        $this->em->persist($b);
+        $this->em->persist($c);
+        $this->em->persist($d);
+        $this->em->flush();
+        $this->assertTrue($b->getId() < $d->getId());
+
+        $excludedIds = array();
+        $this->assertSame($d, $this->repo->findStartableJob($excludedIds, array(), array('default')));
+        $this->assertEmpty($excludedIds);
```

Comment:
Missing a check on object being correctly loaded (since we use the `ResultSetMappingBuilder`)